src/main/scala/xiangshan/backend/issue/Dispatch2Iq.scala

```diff
@@ -139,6 +142,351 @@ abstract class Dispatch2IqImp(override val wrapper: Dispatch2Iq)(implicit p: Par
+    IQ1Enq0Select := Cat(false.B, false.B, false.B, false.B).asBools
+    IQ1Enq1Select := Cat(false.B, false.B, false.B, false.B).asBools
+  }.elsewhen(uopsOutDq0.last.ready){
+    IQ0Enq0Select := Cat(false.B, false.B, false.B, false.B).asBools
+    IQ0Enq1Select := Cat(false.B, false.B, false.B, false.B).asBools
+    IQ1Enq0Select := Cat(false.B, false.B, false.B, true.B).asBools
+    IQ1Enq1Select := Cat(false.B, false.B, true.B, false.B).asBools
+  }.otherwise{
+    IQ0Enq0Select := Cat(false.B, false.B, false.B, false.B).asBools
+    IQ0Enq1Select := Cat(false.B, false.B, false.B, false.B).asBools
+    IQ1Enq0Select := Cat(false.B, false.B, false.B, false.B).asBools
+    IQ1Enq1Select := Cat(false.B, false.B, false.B, false.B).asBools
+  }
+  assert(PopCount(IQ0Enq0Select) < 2.U, "IQ0Enq0Select is not one hot")
+  assert(PopCount(IQ0Enq1Select) < 2.U, "IQ0Enq1Select is not one hot")
+  assert(PopCount(IQ1Enq0Select) < 2.U, "IQ1Enq0Select is not one hot")
+  assert(PopCount(IQ1Enq1Select) < 2.U, "IQ1Enq1Select is not one hot")
+  assert(PopCount(IQ0Enq0Select) < 2.U, "IQ0Enq0Select is not one hot")
+  assert(PopCount(IQ0Enq1Select) < 2.U, "IQ0Enq1Select is not one hot")
+  assert(PopCount(IQ1Enq0Select) < 2.U, "IQ1Enq0Select is not one hot")
+  assert(PopCount(IQ1Enq1Select) < 2.U, "IQ1Enq1Select is not one hot")
+  assert(PopCount(VecInit(IQ0Enq0Select(0), IQ0Enq1Select(0), IQ1Enq0Select(0), IQ1Enq1Select(0))) < 2.U, "Dq1Uop0Select is not one hot")
+  assert(PopCount(VecInit(IQ0Enq0Select(1), IQ0Enq1Select(1), IQ1Enq0Select(1), IQ1Enq1Select(1))) < 2.U, "Dq1Uop1Select is not one hot")
+  assert(PopCount(VecInit(IQ0Enq0Select(2), IQ0Enq1Select(2), IQ1Enq0Select(2), IQ1Enq1Select(2))) < 2.U, "Dq1Uop2Select is not one hot")
+  assert(PopCount(VecInit(IQ0Enq0Select(3), IQ0Enq1Select(3), IQ1Enq0Select(3), IQ1Enq1Select(3))) < 2.U, "Dq1Uop3Select is not one hot")
+  val dq0Block = Wire(Vec(uopsInDq0Num, Bool()))
+  if (IQ0FuCfgs.size == IQ1FuCfgs.size && IQ0FuCfgs.size == IQ01BothfuCfgs.size) {
+    dq0Block := Cat(false.B, false.B, false.B, false.B).asBools
+  }
+  else {
+    val fuTypes = uopsInDq0.map(_.bits.fuType)
+    val iqNotReady0 = (FuTypeOrR(fuTypes(0),IQ0OnlyFuTypes) && !uopsOutDq0.head.ready) || (FuTypeOrR(fuTypes(0),IQ1OnlyFuTypes) && !uopsOutDq0.last.ready)
+    val iqNotReady1 = (FuTypeOrR(fuTypes(1),IQ0OnlyFuTypes) && !uopsOutDq0.head.ready) || (FuTypeOrR(fuTypes(1),IQ1OnlyFuTypes) && !uopsOutDq0.last.ready)
+    val iqNotReady2 = (FuTypeOrR(fuTypes(2),IQ0OnlyFuTypes) && !uopsOutDq0.head.ready) || (FuTypeOrR(fuTypes(2),IQ1OnlyFuTypes) && !uopsOutDq0.last.ready)
+    val iqNotReady3 = (FuTypeOrR(fuTypes(3),IQ0OnlyFuTypes) && !uopsOutDq0.head.ready) || (FuTypeOrR(fuTypes(3),IQ1OnlyFuTypes) && !uopsOutDq0.last.ready)
+    val conflict2 = (IQ0Enq0Select(2) || IQ0Enq1Select(2) && FuTypeOrR(fuTypes(2),IQ1OnlyFuTypes)) || (IQ1Enq0Select(2) || IQ1Enq0Select(2) && FuTypeOrR(fuTypes(2),IQ0OnlyFuTypes))
```

Comment:
IQ1Enq0Select(2) appears twice here. PLZ explain the meaning of conflict2 and conflict3